packages/package.go

```diff
@@ -401,6 +401,29 @@ func (p *Package) WorksWithCapabilities(capabilities []string) bool {
 	return true
 }
 
+func (p *Package) HasCompatibleSpec(specMin, specMax, kibanaVersion *semver.Version) (bool, error) {
+	if specMin == nil && specMax == nil {
+		return true, nil
+	}
+
+	constraints := []string{}
+	if specMin != nil {
+		constraints = append(constraints, fmt.Sprintf(">=%s", specMin.String()))
+	}
+	if specMax != nil {
+		constraints = append(constraints, fmt.Sprintf("<=%s", specMax.String()))
+	}
+
+	fullConstraint := strings.Join(constraints, ",")
+	constraint, err := semver.NewConstraint(fullConstraint)
+	if err != nil {
+		return false, fmt.Errorf("cannot create constraint %s: %w", fullConstraint, err)
+	}
+
+	formatVersion := semver.MustParse(p.FormatVersion)
```

Comment:
Nit. Maybe we should pre-parse these versions now that they are going to be more used.

Reply:
Updated in https://github.com/elastic/package-registry/pull/1058